gcsfs/core.py

```diff
@@ -458,10 +458,19 @@ def buckets(self):
 
     @classmethod
     def _process_object(self, bucket, object_metadata):
-        object_metadata["size"] = int(object_metadata.get("size", 0))
-        object_metadata["path"] = posixpath.join(bucket, object_metadata["name"])
+        """Process object resource into gcsfs object information format.
+        
+        Process GCS object resource via type casting and attribute updates to
+        the cache-able gcsf object information format. Returns an updated copy
+        of the object resource.
 
-        return object_metadata
+        (See https://cloud.google.com/storage/docs/json_api/v1/objects#resource)
+        """
+        result = dict(object_metadata)
```

Comment:
Why this, isn't it already a dict?

Reply:
The eliminates the side-effect of processing the source object by, instead, creating a shallow copy of the object data. This eliminates the confusing logic you identified in the original bug, in which the function was being called for its side effect but the return value was being discarded.

Reply:
OK, got it. +1 from me.